geary-papermc-spawning/src/main/kotlin/com/mineinabyss/geary/papermc/spawning/config/SpreadSpawnConfig.kt

```diff
@@ -0,0 +1,37 @@
+package com.mineinabyss.geary.papermc.spawning.config
+
+import com.charleskorn.kaml.YamlComment
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+@Serializable
+class SpreadSpawnConfig(
+    @YamlComment("Entity type, e.g. 'mm:prayingskeleton'")
+    val entityType: String,
+    @YamlComment("How far the entity should be from another entity of the same type")
+    val spreadRadius: Int,
+    val spawnCap: Int,
+    //val spawnTickDelay: Int, // can't really change that since its handled by the task
+    @YamlComment("Lowest point of the section to spawn in")
+    val sectionMinY: Int,
+    @YamlComment("Highest point of the section to spawn in")
+    val sectionMaxY: Int,
+    @YamlComment("How many chunks the selected area should deviate from the chosen area at most")
+    val spawnNoise: Int,
+    @YamlComment("The size of each 'split' we do when spawning entities, higher means more performance but less precision")
+    val splitSize: Int,
+    @YamlComment("How many times to try and spawn an entity in a chunk before giving up")
+    val spawnAttempts: Int,
+    @YamlComment("Radius around the selected block to be valid")
+    val openAreaWidth: Int,
+    @YamlComment("Height of the open area around the selected block to be valid")
+    val openAreaHeight: Int,
+) {
```

Comment:
I mention earlier reusing geary's conditions system, you could take a list of spawn entries here for allowed spawns in that region and check against them.

In my eyes though, these are two separate configs. One is for the regions, where you define everything that you have above except the entity-specific stuff like entity type, spread radius, and cap.
The second is individual files for each entity type, which just references the section it can spawn it, similar to how the current spawns work.

The latter can wait for now since we only have a handful of skeleton types, but you might imagine in the future if we had 10 different entities we wanted to spread-spawn and 15 regions, this config file would get pretty huge.